### PR TITLE
Add missing probes

### PR DESCRIPTION
**Pull request by skonto**
# Changes
- Add probes to net-istio-controller and net-istio-webhook

### CODE REVIEW

File: cmd/webhook/main.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func main() {
 	})
 
 	sharedmain.WebhookMainWithContext(
-		ctx, "net-istio-webhook",
+		sharedmain.WithHealthProbesDisabled(ctx), "net-istio-webhook",
```
Comment (skonto) [inline cmd/webhook/main.go:88]: Deactivating the default endpooint as we will use the webhook one.

File: config/500-webhook-deployment.yaml [hunk 2]
```diff
@@ -82,6 +82,23 @@ spec:
...
+              - name: k-kubelet-probe
+                value: "webhook"
+        livenessProbe:
+          periodSeconds: 1
+          httpGet:
+            scheme: HTTPS
+            port: 8443
+            httpHeaders:
+              - name: k-kubelet-probe
+                value: "webhook"
```
Comment (nak3) [inline config/500-webhook-deployment.yaml:100]: Should we have same values with serving's webhook (i.e. should not be same between readiness and liveness )? - https://github.com/knative/serving/blob/main/config/core/deployments/webhook.yaml#L122-L123

Here are the discussion:
* https://github.com/knative/serving/pull/9215
* https://github.com/knative/serving/issues/9077

File: cmd/webhook/main.go (hunk 1 above)
Comment (nak3) [reply cmd/webhook/main.go:88]: Although it makes sense, do you know why serving's webhook is not disabling it ? - https://github.com/knative/serving/blob/main/cmd/webhook/main.go#L166-L171

**Review by ReToCode — COMMENTED** [review]
Good points @nak3.
Other than that, LGTM

File: cmd/webhook/main.go (hunk 1 above)
Comment (skonto) [reply cmd/webhook/main.go:88]: I added the default probes at the knative/pkg side and it was suggested to enable by default :shrug:. I guess I can do a follow up there too.

File: config/500-webhook-deployment.yaml (hunk 2 above)
Comment (skonto) [reply config/500-webhook-deployment.yaml:100]: Yes I am aware of that was not sure to what extend it is a problem.

**Comment by skonto** [comment]
@nak3 I changed the threshold values I will do a PR at the Serving side too.

File: config/500-controller.yaml [hunk 3]
```diff
@@ -81,11 +81,28 @@ spec:
           seccompProfile:
             type: RuntimeDefault
 
+        readinessProbe:
+          httpGet:
+            path: /readiness
+            port: probes
+            scheme: HTTP
+          periodSeconds: 5
+          failureThreshold: 3
```
Comment (skonto) [inline config/500-controller.yaml:90]: default is 3 but I want to make it explicit.

**Comment by skonto** [comment]
@nak3 I have created this PR https://github.com/knative/serving/pull/13885 to address the Serving issue. We should backport downstream when all these are merged.

**Comment by nak3** [comment]
Thank you. Let's wait until serving's PR was merged. We can apply the feedback into net-istio as well if necessary.

**Comment by dprotaso** [comment]
FYI Serving's PR merged

File: config/500-webhook-deployment.yaml [hunk 4]
```diff
@@ -82,6 +82,25 @@ spec:
...
+          failureThreshold: 6
+
```
Comment (nak3) [inline config/500-webhook-deployment.yaml:103]: One more thing, probably we should add `initialDelaySeconds: 20`? I think https://github.com/knative/serving/pull/9661 makes sense and other webhook have it.

**Comment by nak3** [comment]
/lgtm
/approve

Thank you!

**Pull request merged**